src/native/libs/System.Security.Cryptography.Native.Android/pal_x509.c

```diff
@@ -346,3 +346,28 @@ static void FindCertStart(const uint8_t** buffer, int32_t* len)
         }
     }
 }
+
+jobject /*X509Certificate*/ AndroidCryptoNative_X509GetCertificateForPrivateKeyEntry(jobject /*PrivateKeyEntry*/ privateKeyEntry)
+{
+    abort_if_invalid_pointer_argument (privateKeyEntry);
+
+    JNIEnv* env = GetJNIEnv();
+
+    jobject cert = (*env)->CallObjectMethod(env, privateKeyEntry, g_PrivateKeyEntryGetCertificate);
+    if (CheckJNIExceptions(env) || !cert)
+    {
+        return NULL;
+    }
+
+    return ToGRef(env, cert);
+}
+
+bool AndroidCryptoNative_X509IsKeyStorePrivateKeyEntry(jobject entry)
+{
+    if (!entry)
+        return false;
+
+    JNIEnv* env = GetJNIEnv();
+
+    return (*env)->IsInstanceOf(env, entry, g_PrivateKeyEntryClass);
+}
```

Comment:
A C `bool` is not typically used across .NET marshaling boundaries. Consider returning it as an `int32_t` and changing the LibraryImport definition to `[return: MarshalAs(UnmanagedType.Bool)]`

_Wait, Kevin, change it to an int here and a bool there?_

Yeah. UnmanagedType.Bool is always 32-bits. So it would look something like:
```suggestion
int32_t AndroidCryptoNative_X509IsKeyStorePrivateKeyEntry(jobject entry)
{
    if (!entry)
        return false;

    JNIEnv* env = GetJNIEnv();

    return (*env)->IsInstanceOf(env, entry, g_PrivateKeyEntryClass) ? 1 : 0;
}
```

Of course you would need to make sure the p/invoke matches, and change the header.